packages/headless/src/api/search/search/search-response.ts

```diff
@@ -15,7 +15,7 @@ import {SecurityIdentity} from './security-identity';
 import {PhrasesToHighlight, TermsToHighlight} from './stemming';
 
 export interface SearchResponseSuccess {
-  generateAutomaticFacets: AutomaticFacets;
+  generateAutomaticFacets?: AutomaticFacets;
```

Comment:
Unrelated change :  I introduced a bug last PR. this and the changes in `headless-automatic-facet-builder.ts` fixes it.